src/router/storeSync/layersParamParser.js

```diff
@@ -50,12 +50,14 @@ export function decodeUrlLayerId(urlLayerId) {
         // KML/GPX do not have layer IDs, so we use their baseUrl as "ID"
         const decodedLayerId = layerId ? decodeExternalLayerParam(layerId) : decodedBaseUrl
         return {
+            isExternal: true,
             id: decodedLayerId,
             type: layerType,
             baseUrl: decodedBaseUrl,
         }
     }
     return {
+        isExternal: false,
```

Comment:
We need to take care of the definition of `isExternal` ! In the layer object (e.g. WMTSLayer, KMLLayer, ...) isExternal has a slightly different meaning which will lead to errors in future or even now ! Here you take every layer that has a type aka starting with `LAYER_TYPE|` as external while in the layer abstract KML layers generated by the application will be an internal even thought it starts with `KML|` ! I would not add this flag here just for the validation purpose. For the validation simply check for the types and with your filter for layer config, you must have a `baseUrl` to check for so no need for this `isExternal`